src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/GroupCollection.cs

```diff
@@ -241,6 +289,20 @@ public Group Current
                 }
             }
 
+            KeyValuePair<string, Group> IEnumerator<KeyValuePair<string, Group>>.Current
+            {
+                get
+                {
+                    if (_index < 0 || _index >= _collection.Count)
+                        throw new InvalidOperationException(SR.EnumNotStarted);
+
+                    Group value = _collection[_index];
+
+                    return new KeyValuePair<string, Group>(value.Name, value);
+
+                }
+            }
```

Comment:
Why is this and the additional interface implementation necessary?

Reply:
@stephentoub because IReadOnlyDictionary requires a GetEnumerator that returns an enumerator that implements this interface

Reply:
Ah. Ok.